Building on Arch

I'm trying to build this on Arch linux, the trouble is I'm getting an error, which maybe related to compiler options of either
d_main.c:2472:31: error: expected ‘)’ before ‘PACKAGE_DOOM_VERSION’
 2472 |     DEH_printf("Russian Doom " PACKAGE_DOOM_VERSION);
      |               ~               ^~~~~~~~~~~~~~~~~~~~~
      |                               )
make[3]: *** [Makefile:539: d_main.o] Error 1
when running the autogen method of compiling or:
d_main.c:1280:34: error: expected ‘)’ before ‘PACKAGE_HERETIC_VERSION’
 1280 |     DEH_printf("Russian Heretic " PACKAGE_HERETIC_VERSION);
      |               ~                  ^~~~~~~~~~~~~~~~~~~~~~~~
      |                                  )
when running cmake,
I'm fairly sure I've got all the dependencies, this just seems to be an error when trying to compile using the function as written in deh_str.c
Is this something other people are seeing? Or have I missed a dependency somewhere?

Hello, thanks for report!

Can reproduce it under Linux, but not happening under Windows/MSYS. Apparently, there are two problems:

1) Missing comma between `"Russian Doom "` and `PACKAGE_DOOM_VERSION`. Once it added, this leading to second problem.
2) 
```
d_main.c: In function ‘D_DoomMain’:
d_main.c:2472:33: error: ‘PACKAGE_DOOM_VERSION’ undeclared (first use in this function); did you mean ‘PACKAGE_VERSION’?
 2472 |     DEH_printf("Russian Doom ", PACKAGE_DOOM_VERSION);
      |                                 ^~~~~~~~~~~~~~~~~~~~
      |                                 PACKAGE_VERSION
```

@Dasperal could you please help us?

--- a/src/doom/d_main.c
+++ b/src/doom/d_main.c
@@ -2469,7 +2469,7 @@ void D_DoomMain (void)
 #else
     // [JN] Just print an uncolored banner
     for (p = 0 ; p < 32 ; p++) DEH_printf(" ");
-    DEH_printf("Russian Doom " PACKAGE_DOOM_VERSION);
+    DEH_printf("Russian Doom " BUILD_DOOM_VERSION);
     for (p = 0 ; p < 31 ; p++) DEH_printf(" ");
     DEH_printf("\n");
 #endif

--- a/src/heretic/d_main.c
+++ b/src/heretic/d_main.c
@@ -1277,7 +1277,7 @@ void D_DoomMain(void)
 #else
     // [JN] Just print an uncolored banner
     for (p = 0 ; p < 32 ; p++) DEH_printf(" ");
-    DEH_printf("Russian Heretic " PACKAGE_HERETIC_VERSION);
+    DEH_printf("Russian Heretic " BUILD_HERETIC_VERSION);
     for (p = 0 ; p < 31 ; p++) DEH_printf(" ");
     DEH_printf("\n");
 #endif 

--- a/src/hexen/h2_main.c
+++ b/src/hexen/h2_main.c
@@ -494,7 +494,7 @@ void D_DoomMain(void)
 #else
     // [JN] Just print an uncolored banner
     for (p = 0 ; p < 32 ; p++) ST_Message(" ");
-    ST_Message("Russian Hexen " PACKAGE_HEXEN_VERSION);
+    ST_Message("Russian Hexen " BUILD_HEXEN_VERSION);
     for (p = 0 ; p < 31 ; p++) ST_Message(" ");
     ST_Message("\n");
 #endif

